Assets/AXIP/AILIA-MODELS/ImageSegmentation/SegmentAnything2Model.cs

```diff
@@ -0,0 +1,1863 @@
+        for (int y = 0; y < imageHeight; y++)
+        {
+            int unityY = y;
+            int rowOffset = unityY * imageWidth;
+
+            for (int x = 0; x < imageWidth; x++)
+            {
+                int pixelIndex = rowOffset + x;
+
+                if (pixelIndex >= 0 && pixelIndex < pixels.Length)
+                {
+                    Color32 originalColor = pixels[pixelIndex];
+                    pixels[pixelIndex] = originalColor;
+                }
+            }
+        }
+
+        result.SetPixels32(pixels);
+        result.Apply();
+        return result;
+    }
+
+    // Visualize clicked points
+    private Texture2D DrawClickPoints(float[,] coords, float[] labels, Texture2D image)
+    {
+        Texture2D result = new Texture2D(image.width, image.height, image.format, false);
+        Graphics.CopyTexture(image, result);
+        Color32[] pixels = result.GetPixels32();
+
+        int numPoints = coords.GetLength(0);
+        int markerSize = 15;
+
+        for (int i = 0; i < numPoints; i++)
+        {
+            if (labels[i] >= 2)
+            {
+                continue;
+            }
+
+            int px = Mathf.Clamp((int)coords[i, 0], 0, image.width - 1);
+            int origY = (int)coords[i, 1];
+
+            // Convert to Unity coordinates
+            int py = origY;
+            py = Mathf.Clamp(py, 0, image.height - 1);
+
+            Color32 markerColor =
+                labels[i] == 1 ? new Color32(0, 255, 0, 255) : new Color32(0, 0, 255, 255);
+
+            //Debug.Log($"{coords[i, 0]},{coords[i, 1]} => {px}, {py}");
+
+            // Draw marker with bounds checking in the loop
+            for (int dy = -markerSize; dy <= markerSize; dy++)
+            {
+                for (int dx = -markerSize; dx <= markerSize; dx++)
+                {
+                    if (Math.Abs(dx) == Math.Abs(dy))
+                    {
+                        int nx = px + dx;
+                        int ny = py + dy;
+
+                        if (nx >= 0 && nx < image.width && ny >= 0 && ny < image.height)
+                        {
+                            int idx = ny * image.width + nx;
+                            pixels[idx] = markerColor;
+                        }
+                    }
+                }
+            }
+        }
+
+        result.SetPixels32(pixels);
+        result.Apply();
+        return result;
+    }
+
+    private float GetScale(int texWidth, int texHeight)
+    {
+        return targetSize / (float)Mathf.Max(texWidth, texHeight);
+    }
+
+    private float[,,,] PreprocessImage(
+        Color32[] inputImage,
+        int originalWidth,
+        int originalHeight,
+        int imageSize
+    )
+    {
+        // Resize image to (imageSize, imageSize)
+        float scale = GetScale(originalWidth, originalHeight);
+        int newWidth = imageSize;
+        int newHeight = imageSize;
+        Color32[] resizedImage = ResizeColor32(
```

Comment:
In ResizeColor32, scaling is performed without maintaining the aspect ratio, ignoring scale. In ApplyCoordinateScaling, the coordinates of the Point are corrected with scale. A mismatch occurs here. Please provide the scale as an argument to ResizeColor32 and perform resizing while maintaining the aspect ratio.